cmd/kyma/alpha/deploy/cmd.go

```diff
@@ -39,51 +42,66 @@ func NewCmd(o *Options) *cobra.Command {
 	cobraCmd.Flags().DurationVarP(&o.HelmTimeout, "helm-timeout", "", 360*time.Second, "Timeout for the underlying Helm client.")
 	cobraCmd.Flags().IntVar(&o.WorkersCount, "workers-count", 4, "Number of parallel workers used for the deployment.")
+	cobraCmd.Flags().StringVarP(&o.Domain, "domain", "d", o.getDefaultDomain(), "Domain used for installation.")
+	cobraCmd.Flags().StringVarP(&o.TLSCert, "tls-cert", "", "", "TLS certificate for the domain used for installation. The certificate must be a base64-encoded value.")
+	cobraCmd.Flags().StringVarP(&o.TLSKey, "tls-key", "", "", "TLS key for the domain used for installation. The key must be a base64-encoded value.")
+	cobraCmd.Flags().StringVarP(&o.Version, "source", "s", o.getDefaultVersion(), `Installation source. 
+	- To use a latest release, write "kyma alpha deploy --source=latest".
+	- To use a specific release, write "kyma alpha deploy --source=1.17.1".
+	- To use the master branch, write "kyma alpha deploy --source=master".
+	- To use a commit, write "kyma alpha deploy --source=34edf09a".
+	- To use a pull request, write "kyma alpha deploy --source=PR-9486".
+	- To use the local sources, write "kyma alpha deploy --source=local".`)
+	cobraCmd.Flags().StringVarP(&o.Profile, "profile", "p", "",
+		fmt.Sprintf("Kyma deployment profile. Supported profiles are: %s", strings.Join(o.getProfiles(), ", ")))
 	return cobraCmd
 }
 
 //Run runs the command
 func (cmd *command) Run() error {
 	var err error
-	if err = cmd.validateFlags(); err != nil {
+
+	// verify input parameters
+	if err = cmd.opts.validateFlags(); err != nil {
 		return err
 	}
 
+	// initialize Kubernetes client
 	if cmd.K8s, err = kube.NewFromConfig("", cmd.KubeconfigPath); err != nil {
 		return errors.Wrap(err, "Could not initialize the Kubernetes client. Make sure your kubeconfig is valid")
 	}
 
-	var componentsContent string
-	if cmd.opts.ComponentsYaml != "" {
-		data, err := ioutil.ReadFile(cmd.opts.ComponentsYaml)
+	// initialize UI
+	var updateCh chan deployment.ProcessUpdate
+	if cmd.Verbose {
+		defer log.Println("Kyma deployed!")
+	} else {
+		asyncUI := asyncui.AsyncUI{StepFactory: &cmd.Factory}
+		updateCh, err = asyncUI.Start()
 		if err != nil {
-			return fmt.Errorf("Failed to read installation CR file: %v", err)
+			return err
 		}
-		componentsContent = string(data)
+		defer asyncUI.Stop() // stop receiving update-events and wait until UI rendering is finished
 	}
 
-	var overridesContent []string
-	if cmd.opts.OverridesYaml != "" {
-		data, err := ioutil.ReadFile(cmd.opts.OverridesYaml)
-		if err != nil {
-			return fmt.Errorf("Failed to read installation CR file: %v", err)
-		}
-		overridesContent = append(overridesContent, string(data))
+	//to add another stop to the UI, just call:
+	if result, _ := cmd.isK3sCluster(); result {
```

Comment:
This is just example code to explain how additional steps can be shown in the async-UI. This block will be removed in the next PR.